web/packages/shared/components/DesktopSession/InputHandler.tsx

```diff
@@ -118,6 +134,77 @@ export class InputHandler {
     }
   }
 
+  /**
+   * Updates the locally tracked remote modifier state.
+   *
+   * @param keyCode The key code of the key.
+   * @param state The state of the key.
+   */
+  private updateModifierState(keyCode: string, state: ButtonState) {
+    switch (keyCode) {
+      case 'ShiftLeft':
+      case 'ShiftRight':
+        this.remoteModifierState.set('Shift', state);
+        break;
+      case 'ControlLeft':
+      case 'ControlRight':
+        this.remoteModifierState.set('Control', state);
+        break;
+      case 'AltLeft':
+      case 'AltRight':
+        this.remoteModifierState.set('Alt', state);
+        break;
+      case 'MetaLeft':
+      case 'MetaRight':
+        this.remoteModifierState.set('Meta', state);
+        break;
+    }
+  }
+
+  /**
+   * Checks if the given key is a modifier key.
+   *
+   * @param key The key to check.
+   * @returns True if the key is a modifier key, false otherwise.
+   */
+  private isModifierKey(key: string): boolean {
+    return ['Shift', 'Control', 'Alt', 'Meta'].some(modifier =>
+      key.startsWith(modifier)
+    );
+  }
+
+  /**
+   * Synchronizes the local modifier state with the remote machine.
+   * This is called when a key event occurs to ensure that the remote machine
+   * has the correct state of modifier keys. If not, it sends the current state.
```

Comment:
The comment needs to be updated since it's no longer called only on a key event.